### ISSUE TITLE
Usability feedback

### ISSUE THREAD

**Issue by DavidIQ** [phpbb-extensions/phpbb-ext-skeleton#95]
Some comments on usability after generating an extension with the latest changes:

- All CP migration files are being generated even if they aren't all selected:
![image](https://user-images.githubusercontent.com/186341/51329016-93b83f00-1a42-11e9-9d2b-02d0466ea59a.png)
![image](https://user-images.githubusercontent.com/186341/51328947-72575300-1a42-11e9-95d6-4c43b51498a5.png)
- ACP language files are being generated with the package name for the key values instead of the display name.
- `depends_on` for migrations assumes 3.1 (`v31x\v314`) even though minimum is automatically being set to 3.2.0 on the interface.

On that last one there seems to be some sort of desire to keep 3.1 support in this extension even though 3.1 has been EoL since January 2018. All 3.1 support in this extension should be retired in the next release.

**Comment by VSEphpbb** [comment]
All 3.1 support really means in this extension at this point, is that the User object will still handle the language methods, and a few other minor things for 3.1 backwards compaibility.

I don't want to shut people out of the option of generating 3.1 BC compatible skeletons. The CDB would not deny an extension for still working in both 3.1 and 3.2 (i.e.: using the User object), nor would it deny an extension whose minimum phpBB requirement is 3.1.0 (as defined in their composer.json).

Basically, if an author wants their extension minimum phpBB version supported to be 3.1.0, then this extension should generate an appropriate skeleton for that.

**Comment by VSEphpbb** [comment]
Can you expand on your first point? I'm not seeing that.
Nothing selected:
![screen shot 2019-01-16 at 10 48 15 pm](https://user-images.githubusercontent.com/303711/51300492-ecef8680-19e0-11e9-9500-30de18ab24d9.png)

Only MCP selected:
![screen shot 2019-01-16 at 10 48 29 pm](https://user-images.githubusercontent.com/303711/51300497-f37dfe00-19e0-11e9-917b-c1123e0b11d0.png)

**Comment by VSEphpbb** [comment]
As to the 2nd point, it's intended to use the packge name in keys and not display name.

We want keys like: `FOO_BAR_KEY`
We don't want keys like: `WACKY_WAVING_INFLATABLE_FLAILIANG_ARM_TUBE_MAN_KEY`

The extension package name will be short, sweet and free from special characters and is the best choice, as opposed to the more verbose and unrestricted display names. 😉

**Comment by DavidIQ** [comment]
For the first point I had selected ACP but ended up with all CPs. I'll have to go back and try generating again as there must be some other options I selected to cause that.

For the second point note how I said the value (right side) and not the key (left side). What is generated looks something like this:
```
'MYEXTENSION_MAIN' => 'myextension Main'
```

**Comment by VSEphpbb** [comment]
Oh. Well even so those ACP Lang values are intentionally like that. The description name is still not as appropriate as the extension name for things like a Module name in our generic skeleton. Obviously users will should be changing them to suit their needs.

**Comment by DavidIQ** [comment]
How is this (package name):
```
'MYEXTENSION_MAIN' => 'myextension Main'
```
more appropriate than this (display name):
```
'MYEXTENSION_MAIN' => 'My Extension Main'
```

**Comment by VSEphpbb** [comment]
Again, the naming does not truly matter since users should be changing things to what they really need.

The naming comes from what the Skeleton was in it's previous version. Anywhere Acme was used we went with VENDOR, anuwhere DEMO was used, we went with PACKAGE and anywhere Acme Demo was used, we went with DESCRIPTION NAME.
https://github.com/phpbb-extensions/phpbb-ext-skeleton/commit/0cb2b29aecb925211cec6b8be81427e46de3a1b8#diff-acc00e952521ae31ec72a7c5a151e00fL23

**Comment by VSEphpbb** [comment]
Currently when you choose migrations, all migration files are generated. There may be no need for the migration check box since every migration file can also be generated by choosing another object, like ACP, MCP or UCP, however the idea was if you want sample migrations check this box and you'll get a bunch of examples.

**Comment by VSEphpbb** [comment]
Maybe we could make a generic and useless migration file specifically for that checkbox. And let all the other migration files just come from their respective CP module checkboxes.

**Comment by 3D-I** [comment]
I actually complained about this years ago, https://github.com/phpbb-extensions/phpbb-ext-skeleton/issues/39 . I agree with the all the points made by David.

**Comment by DavidIQ** [comment]
> Again, the values do not truly matter since users should be changing things to what they really need.
> 
> The naming comes from what the Skeleton was in its previous version. Anywhere Acme was used we went with VENDOR, anywhere DEMO was used, we went with PACKAGE and anywhere Acme Demo was used, we went with DESCRIPTION NAME.
> [0cb2b29#diff-acc00e952521ae31ec72a7c5a151e00fL23](https://github.com/phpbb-extensions/phpbb-ext-skeleton/commit/0cb2b29aecb925211cec6b8be81427e46de3a1b8#diff-acc00e952521ae31ec72a7c5a151e00fL23)

This doesn't make a whole lot of sense. You're wanting to force the user to _always_ replace one thing with another that they've already entered? Why? Just because you want them to consciously do that? Every time I generate an extension with an ACP module I'm having to go into the language file and replace the log key's value, the title's value, and the settings' value to exactly what I entered for the title. Same for some of the other generated language files. I can't imagine that I'm in the minority in having to do this pointless exercise for every generated extension.

> Maybe we could add a generic and useless migration file specifically for that checkbox. And let all the other migration files just come from their respective CP module checkboxes. One that does some basic stuff like add a config, config_text, etc.

We already have a useless migration example with the user_schema one. The CP ones are more like a necessity when those modules are selected. At this point this extension should not be considered a learning tool and more in line with what the README states:
> phpBB's Skeleton Extension is a tool for extension authors to help speed up and simplify the task of starting a new extension project. It generates sample starter files and directories in a skeleton package that you can use to begin building your extension.

While it is true that the user could just delete the files, as stated in @3D-I's issue, the whole point of this issue is to improve usability.

**Comment by VSEphpbb** [comment]
> This doesn't make a whole lot of sense. You're wanting to force the user to always replace one thing with another that they've already entered? Why? Just because you want them to consciously do that? Every time I generate an extension with an ACP module I'm having to go into the language file and replace the log key's value, the title's value, and the settings' value to exactly what I entered for the title. Same for some of the other generated language files. I can't imagine that I'm in the minority in having to do this pointless exercise for every generated extension.

Well of course that's what we want! How can we assume what the user is actually trying to do???? The whole point is this is a boiler plate with very minimal functional code for example/informational purposes. Things like lang keys and values are named generically, because they need to be named something, and the user can see how they are being used. This extension is not and has never been designed to build your extensions for you...just to get you a batch of generic files to get off and running writing your own code.

Originally, until this update, what it output was NickV's Acme Demo extension (which meant searching for `acme` and `demo` throughout all the files to change to your own naming). In this update we simply wanted to remove all the Acme and Demo references with the user's own vendor and package name since, in some case, but not all, those names will fall into place and not need changing (such as in namespaces, for example). But of course a user will need to -- I don't know -- build their own extension 😉  and that means removing the sample code they don't need and modifying the remaining code that works for them to their own needs.

**Comment by DavidIQ** [comment]
That's fine and dandy, except you also replaced all instances of `Acme`, which is what is displayed on the interface, with `whateverextension` which was never displayed. You can argue all you want about it, but that was done wrong and now needs fixing.

**Comment by VSEphpbb** [comment]
Also, as you pointed out:

>It generates sample starter files

The keyword there being "sample" 😄 

I don't know what you mean by `Acme` which is displayed on the interface...or what was actually done wrong?
